### ISSUE TITLE
MSF update error

### ISSUE THREAD

**Issue by Systemad** [cSploit/android#43]
So I get "Error occurred during update" or something like that when I download MSF(the msg pop-up after it gets 100% on "extracting" and every time istart the application I get toast msg "cannot start process" (I believe it's #1), I also tried with the new latest pre-release and same error occurred.
Here is my logcat : https://ghostbin.com/paste/8j6x8 (device info can be found here also) 
and I got no csploitd.log :/

**Comment by DNBkid** [comment]
i have the same issue , what model is your device ?

**Comment by Systemad** [comment]
HTC one m7 (ARHD 83.0 4.4.3)

**Comment by tux-mind** [comment]
@Systemad you are not affected by #1 , you can run `which env`.

it seems that your ruby installation is corrupted.

try to *Settings > Delete MSF* and install ruby again.

thanks for the report.

keep me posted.

**Comment by Systemad** [comment]
I already done that like 5 times, the ruby download is very fast(3seconds) I  have heard that it should take more than that:/

**Comment by Systemad** [comment]
Seconds*

**Comment by Systemad** [comment]
I have also tried installing ruby on SD card with no luck

**Comment by tux-mind** [comment]
@Systemad did you installed any other ruby on you device ?

it seems that your ruby version is 1.9.1 while the one shipped with cSploit is 1.9.3

**Comment by Systemad** [comment]
Yeah, I noticed it when I checked data/data/org.csploit.android/(ruby/lib/ruby, then it's a folder named 1.9.1)  
I deleted it, tried reinstall ruby and MSF. Reinstall app. Same happens.

**Comment by Systemad** [comment]
I also tried deleting the whole folder. But it only downloads version 1.9.1. I am not a developer, so I don't know what to do. :/

**Comment by tux-mind** [comment]
@Systemad the 1.9.1 folder is ok, ruby 1.9.3 still use the 1.9.1 folders, don't worry about it.

it seems that ruby it's fine, but the `gem install bundle` command fails with `execvp: permission denied`

i need your help to understand what is happening on your device.

after a successful installation of the ruby package please do the following from the adb shell and report back the output.

```sh

export RUBYLIB=/storage/emulated/0/ruby/lib/ruby/site_ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/site_ruby/1.9.1/arm-linux-androideabi:/storage/emulated/0/ruby/lib/ruby/site_ruby:/storage/emulated/0/ruby/lib/ruby/vendor_ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/vendor_ruby/1.9.1/arm-linux-androideabi:/storage/emulated/0/ruby/lib/ruby/vendor_ruby:/storage/emulated/0/ruby/lib/ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/1.9.1/arm-linux-androideabi
export PATH=/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin:/storage/emulated/0/ruby/bin
export HOME=/storage/emulated/0/ruby/home/ruby
which ruby
which gem
head -n1 $(which gem)
ruby -v
gem -v
gem install bundle
```

feel free to ask me any further info, like how to get in an adb shell.

thanks in advance for your time.

**Comment by DNBkid** [comment]
no i did not, i have installed only csploit and the ruby that came after csploit start updating. 




-------- Оригинално писмо --------
От:  tux-mind  
Относно: Re: [android] MSF update error (#43)
До: cSploit/android  
Изпратено на: Петък, 2015, Януари 16 02:01:12 EET



 @Systemad  did you installed any other ruby on you device ? 


it seems that your ruby version is 1.9.1 while the one shipped with cSploit is 1.9.3

**Comment by Systemad** [comment]
I will do this ASAP but what will happen after I do this? And what should I do? (Open cSploit..?? or?

**Comment by DNBkid** [comment]
here is the catlog and screenshot from adb shell. if i did it wrong please let me know ,to do it again correctly : )




-------- Оригинално писмо --------
От:  tux-mind  
Относно: Re: [android] MSF update error (#43)
До: cSploit/android  
Изпратено на: Вторник, 2015, Януари 20 16:39:07 EET



 @Systemad  the 1.9.1 folder is ok, ruby 1.9.3 still use the 1.9.1 folders, don't worry about it. 


it seems that ruby it's fine, but the  gem install bundle  command fails with  execvp: permission denied  


i need your help to understand what is happening on your device. 


after a successful installation of the ruby package please do the following from the adb shell and report back the output. 

   export  RUBYLIB=/storage/emulated/0/ruby/lib/ruby/site_ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/site_ruby/1.9.1/arm-linux-androideabi:/storage/emulated/0/ruby/lib/ruby/site_ruby:/storage/emulated/0/ruby/lib/ruby/vendor_ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/vendor_ruby/1.9.1/arm-linux-androideabi:/storage/emulated/0/ruby/lib/ruby/vendor_ruby:/storage/emulated/0/ruby/lib/ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/1.9.1/arm-linux-androideabi
 export  PATH=/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin:/storage/emulated/0/ruby/bin
 export  HOME=/storage/emulated/0/ruby/home/ruby
which ruby
which gem
head -n1   $( which gem )  
ruby -v
gem -v
gem install bundle  


feel free to ask me any further info, like how to get in an adb shell. 


thanks in advance for your time.

**Comment by DNBkid** [comment]
Here is my second try, but im still thinking that i do it wrong again.



-------- Оригинално писмо --------
От:  tux-mind  
Относно: Re: [android] MSF update error (#43)
До: cSploit/android  
Изпратено на: Вторник, 2015, Януари 20 16:39:07 EET



 @Systemad  the 1.9.1 folder is ok, ruby 1.9.3 still use the 1.9.1 folders, don't worry about it. 


it seems that ruby it's fine, but the  gem install bundle  command fails with  execvp: permission denied  


i need your help to understand what is happening on your device. 


after a successful installation of the ruby package please do the following from the adb shell and report back the output. 

   export  RUBYLIB=/storage/emulated/0/ruby/lib/ruby/site_ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/site_ruby/1.9.1/arm-linux-androideabi:/storage/emulated/0/ruby/lib/ruby/site_ruby:/storage/emulated/0/ruby/lib/ruby/vendor_ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/vendor_ruby/1.9.1/arm-linux-androideabi:/storage/emulated/0/ruby/lib/ruby/vendor_ruby:/storage/emulated/0/ruby/lib/ruby/1.9.1:/storage/emulated/0/ruby/lib/ruby/1.9.1/arm-linux-androideabi
 export  PATH=/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin:/storage/emulated/0/ruby/bin
 export  HOME=/storage/emulated/0/ruby/home/ruby
which ruby
which gem
head -n1   $( which gem )  
ruby -v
gem -v
gem install bundle  


feel free to ask me any further info, like how to get in an adb shell. 


thanks in advance for your time.

**Comment by Systemad** [comment]
I think i having some problems http://gyazo.com/e64dafddc41f58e6d0f37b4a00cbc8e7

**Comment by tux-mind** [comment]
@DNBkid github drop email attachments, please share a link to the resources or reply from the github web interface.

thanks for your time :)

**Comment by tux-mind** [comment]
@Systemad it seems that you have another ruby installed.

you have ruby installed in `/system/bin/ruby`.

try the commands i sent you before but for the PATH one use the following:

```sh
export PATH=/storage/emulated/0/ruby/bin:/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin
```

thanks for help me kill bugs :)

**Comment by Systemad** [comment]
same thing happens :/ http://gyazo.com/c61c2179e0ff8909276811adde32b709

**Comment by tux-mind** [comment]
there is 2 answers to this behaviour:

  - exevp/which check `/system/bin` before any other directory in `PATH`
  - ruby is not installed in `/storage/emulated/0/ruby/bin`

can you run the following commands ?

```sh
cd /storage/emulated/0/ruby/bin
ls -l
./ruby -v
ls -l /system/bin/ruby
```

thanks for your time.

**Comment by Systemad** [comment]
I might found the problem. to "get in" to the folder the command is /scard/ruby/bin. i did it and here http://gyazo.com/e28152dfa9ce69e310870ef4be3dd3fb

**Comment by Systemad** [comment]
export RUBYLIB=/sdcard/ruby/lib/ruby/site_ruby/1.9.1:/sdcard/ruby/lib/ruby/site_ruby/1.9.1/arm-linux-androideabi:/sdcard/ruby/lib/ruby/site_ruby:/sdcard/ruby/lib/ruby/vendor_ruby/1.9.1:/sdcard/ruby/lib/ruby/vendor_ruby/1.9.1/arm-linux-androideabi:/sdcard/ruby/lib/ruby/vendor_ruby:/sdcard/ruby/lib/ruby/1.9.1:/sdcard/ruby/lib/ruby/1.9.1/arm-linux-androideabi
export PATH=/sdcard/ruby/bin:/sbin:/vendor/bin:/system/sbin:/system/bin:/system/xbin
export HOME=/sdcard/ruby/home/ruby
which ruby
which gem
head -n1 $(which gem) 
/// The commandsi executed

So i guess i fixed it but i get permission denied

**Comment by DNBkid** [comment]
5 minutes ago i decided to uninstall csploit  and to install it again , first ruby updated fine and after metasploit have  finished downloading then for the first time has installed without errors ,but after i restarted the device it gives me again cannot start the process but metasploit is now installed ,i think that we have some progress !  Did you succeeded  with your devise, i have huawei p6 4.4.2

**Comment by DNBkid** [comment]
here is mu catlog of the whole proccess ,also in the catlog i have included the other error i cant start "PASSWOR SNIFFER" 
http://dox.bg/files/dw?a=ee931e6354

**Comment by tux-mind** [comment]
Thank you very much for your patience. I'm quite busy in these months and
my work rate is significantly decreased. My apologies for this.

Thanks for your precious logcat, will give a look tomorrow.

**Comment by Systemad** [comment]
What does this new feature "Network-radar" do? 'I'm sorry if This is not be the best way to contact you? Idk how to contact you otherwise)

**Comment by tux-mind** [comment]
@Systemad my email address `tux_mind at csploit dot org` is not functional.
i want to migrate to amazon ws, but I have to compute how much it will cost to me ( after the 1st free year ).

i've just created a [readme](https://github.com/cSploit/network-radar/blob/master/README.md) for answer your question :heart:

**Comment by tux-mind** [comment]
I've just added support for android 5.

this issue will be the next to get fixed :sunglasses:

**Comment by Ottavio97** [comment]
@tux-mind I meant MSF doesn't start even if it's installed successfully

**Comment by jackzett** [comment]
uhm, tux-mind you asked me for a log, 
look im not sure where to find it, if not deleted it allready, 
i do sure know how delicate the installation process is, doesn't need to be that my self extracting didn't work out thought....

**Comment by Systemad** [comment]
@jackzett
1. install catlog from the play store
2.open it and start recording (in the menu in app)
3.launch cSploit
4. make it show the error
5.tap on catlog notification for stop recording
6.post the file under the /sdcard/catlog directory ( pastebin preferred )

If you want the csploid.log 
/data/data/org.csploit.com/files/csploitd.log

**Comment by alexmanner** [comment]
@tux-mind Great! It doesn't crash on 5.x anymore. This is great! I think that the metasploit framework doesn't work at the moment because there is an error. Please find the logcat in the link below:
http://pastebin.com/WwJRx6Ss

**Comment by tux-mind** [comment]
@alexmanner it's a pleasure to see cSploit run on your lollipop device :blush: 

fixed in 8f7d932bb3c3c00113b1fa9e93cccecdfe0a5330.

please try the nightly build and let me known.
thanks in advance :relaxed:

**Comment by alexmanner** [comment]
@tux-mind Currently, the nightly build is still the version with the heart attack.

**Comment by alexmanner** [comment]
@tux-mind I have seen that you already commented on it in the lollipop thread. :-)

**Comment by Msrpc** [comment]
cSploit not working on SGS3 Lollipop 5.0.2 http://pastebin.com/N34tsZye same error as first post really wish I could get this working :(

**Comment by tux-mind** [comment]
@Msrpc I fixed your error. please go in Settings > Delete MSF.

than update everything to latest version ( a bunch of update dialogs will pop up on startup ).

thanks in advance for your time :blush:

**Comment by KyriakosR** [comment]
@tux-mind I have tried to delete and re-downloaded a million times and it still doesn't work..

**Comment by SaggySally** [comment]
Here is a logcat of the error @tux-mind

Logcat: http://pastebin.com/7vp7HtwK

**Comment by Systemad** [comment]
1 Question, when I installed cSploit it asks to install core 1.0.2, after it install I get "Initializing". How long should it be like that? I Waited 10 min nothing happen s

**Comment by tux-mind** [comment]
@Systemad please go in system settings > applications > cSploit > terminate.
Reopen cSploit.

Will be great if you can provide a logcat.

I've 2 exams on 16/2, so I can't fix it now, sorry.

**Comment by Systemad** [comment]
No problem,
http://pastebin.com/WyLGEqVM
and good luck on your exams! :D

**Comment by jonniex7** [comment]
s3 4.4.2 touchwiz rom QS
http://pastebin.com/kkYZB4nv

**Comment by Msrpc** [comment]
@tux-mind Still not working after complete fresh install. Same error on both my SGS3 CM12 5.0.2 Lollipop and my Asus Memo pad stock Android 4.4.2 

I just don't understand how I'm getting this same error on both... I need help. MetasploitFramework must have some serious bug because I've tried 3 different devices and its always the downloading error for MSF, ruby and core install fine.

New error log:http://pastebin.com/uHmqQMRN

**Comment by Msrpc** [comment]
@tux-mind:Newest error log: http://pastebin.com/UZE8KAqA

Just an extra question which error log type is more useful verbose or debug using catalog? I used debug this time but will reupload with verbose if you need it.

**Comment by DNBkid** [comment]
Here is my catlog  for Csploit Ver. 1.41 
 http://dox.bg/files/dw?a=47018437c4 , 
again i have the same problem with " MetasploitFramework "  it just wont install , can you just remove it and everything will be fine : )))   And good luck with the exams ! ;)

**Comment by tux-mind** [comment]
@Systemad the logcat you provided is when the infinite initialization loop happened ?
if you confirm I'll simulate these steps on my phone for debug this.

From the log I see a successful installation ... no problems.

**Comment by tux-mind** [comment]
@Msrpc I cannot figure out what is going wrong on your device.

can you please share `/data/data/prg.csploit.android/files/cSploitd.log` ??

thanks in advance for your time :blush:

**Comment by orangeisnon** [comment]
By the way, Tux, I also get the "Initializing" forever on first run (whether I choose to install Ruby or not). I don't think it's related to the duplicate Core, though. In fact, my latest set of logs should end like that (after MSF install fails).

**Comment by Msrpc** [comment]
@tux-mind 

Here's the Csploitlog: http://pastebin.com/qfHTsP05

**Comment by tux-mind** [comment]
@DNBkid remove a feature is not the right way to fix a feature's bug :laughing: 

@DNBkid please capture again your logcat with debug infos, I cannot understand what is going bad. you captured it with warnings and errors only.

**Comment by tux-mind** [comment]
@orangeisnon thanks for the hint, I'll give it a look.

**Comment by tux-mind** [comment]
@Msrpc it seems that `gem` is not installed...
can you list the content of the `/data/data/org.csploit.android/files/ruby/bin` directory ?

sorry for making this kind of remote debugging.
I'm trying to reproduce this bug on my phone.

**Comment by Msrpc** [comment]
@tux-mind there is no /bin in the /data/data/org.cSploit.android/files/ruby directory for me :(

**Comment by Msrpc** [comment]
@tux-mind
Contents of /data/data/org.csploit.android/files/ruby 

.nomedia
cSploitd
/handlers
known-issues
start_daemon.sh
/tools
users
VERSION

**Comment by Systemad** [comment]
Yes tuxmind, initialization loop.

**Comment by tux-mind** [comment]
@Msrpc are you sure that what you posted is the content of `/data/data/org.csploit.android/files/ruby` directory ?

it seems the content of `/data/data/org.csploit.android/files/`.

**Comment by Msrpc** [comment]
@tux-mind those are the contents of 
/data/data/org.csploit.android/files/ruby

 I didn't even notice but you are correct that both my ruby folder and files folder have similar or the same content but what would cause that?

**Comment by orangeisnon** [comment]
@Msrpc That's something that Tux and I noticed recently. For whatever reason cSploit is downloading the Core, then when it says it's updatng Ruby, it's actually downloading the Core again and putting it in the Ruby folder. Check for yourself, when installing Ruby, check the notification and see if it says Ruby.xxx.whatever.tar or Core.xxx.whocares.tar.

**Comment by Msrpc** [comment]
@tux-mind So every time I open cSploit it says I have a new MetaSploitFramework update available and it asks me if I want to download it downloads, checking, extracting then installing gems bar loads and then "an error occurred while downloading the update." Pops-up

**Comment by Systemad** [comment]
It says Core.xxx.whocares.tar. for me when I update ruby.

**Comment by orangeisnon** [comment]
This is what happens.
1. Open cSploit (first run)
2. Updates Core
3. States it is updating Ruby, actually installs Core again, this time in the Ruby folder
4. Tries to update MSF, but since Ruby isn't really there, it fails on Gems (which use the Ruby installation, so can't run)
@Msrpc @Systemad , we need to see why it is choosing to install the Core again, rather than Ruby.
Is everyone now on the same page?
( @Systemad I'm in the same boat as you with the Init Loop, but that comes second to the Ruby failure. Let's resolve that first)

**Comment by Msrpc** [comment]
I can also confirm mine says Core.xxx.who cares.tar when installing as well

**Comment by Msrpc** [comment]
@tux-mind Where can I manually download the ruby tar so that I can either manually install it or trick the automatic installation into properly installing the right files.

**Comment by mcilya** [comment]
@msrpc here: /storage/emulated/0/ruby.tar.xz

**Comment by Msrpc** [comment]
@mcilya Can you upload it somewhere or is it online for download? Because cSploit does not download the ruby.tar.xz for me only the core system file.

**Comment by mcilya** [comment]
@Msrpc https://copy.com/JxHo7Ge9OtjKczyR - ruby.tar.xz

**Comment by Msrpc** [comment]
Even when I trick ruby to extract and install the proper files instead of the core when it gets to MSF and games installing the error downloading persists.

**Comment by Msrpc** [comment]
MSF and gems*

**Comment by nitoshi** [comment]
Have installed last nightly build, got same "error during download" after trying to update metasploit.
catlog: http://drp.io/f/oJZ

**Comment by daniloscala** [comment]
great work thanks but i have error in download and update metasploit framework , ruby not all time

**Comment by tux-mind** [comment]
sorry for the late reply, I'm busy with my exams.

working on it.

**Comment by DNBkid** [comment]
" @DNBkid please capture again your logcat with debug infos, I cannot understand what is going bad. you captured it with warnings and errors only.  "  

How to capture with debug ,its not problem to do it , just tell me how : ))

**Comment by tux-mind** [comment]
@DNBkid  [HowTo logcat](http://forum.xda-developers.com/showthread.php?t=1726238).

**Comment by TheRollingEdit** [comment]
Goof luck with your exams. When do you think thid issue is going to be fixed? Really looking forward to using Csploit on my Nexus 6 😀

**Comment by tux-mind** [comment]
It should be fixed in last commit .
Try out the nightly build and let me known.

**Comment by Systemad** [comment]
Not fixed it for me. I  already tried nightly build

**Comment by Skipsterist** [comment]
I've been following and reading for the past few days and decided to sign up to comment. 

This latest nightly resolved the core/ruby situation; it now downloads core, ruby, MSF in that order and extracts the same way. 

However, after the extraction of MSF, the "installing gems for MSF" hangs and doesn't complete. So for me, it goes all the way through except for the gem install. It will sit "working" for about 60 seconds and fails to install.

**Comment by TheRollingEdit** [comment]
Yup, I confirm, it actually installs ruby but the "installing gems" hangs for about 2 mins and then says 'cannot start process" and it asks for a metasploit framework update again, i tried the loop a few times and it doesent end.

**Comment by TheRollingEdit** [comment]
Also, if it can help, if when it asks for the metasploit update i say yes, let it fail, but say no the second time it asks, if i click on any device, and then inspector and start, for 1 frame it says stop and then comes back to start, I basicely cant do anything but the network tree works.

**Comment by Msrpc** [comment]
Mine doesn't ask for a repeat Metasploit update after letting gems finish installing but it still "cannot start process". Got my hopes up though can't wait until it works, can't really use anything without it though.

**Comment by tux-mind** [comment]
should be fixed in c690b4fdcbd89a5f4f0e87daf9c9e63c0659dbdd

I'm building a new version right now...

**Comment by TheRollingEdit** [comment]
Thanks a lot for all you hard work. Everyone thay uses Csploit really appreciates it ☺.

**Comment by tux-mind** [comment]
@TheRollingEdit thanks :blush:

**Comment by tux-mind** [comment]
done, let me known if we can finally close this issue :sunglasses:

**Comment by TheRollingEdit** [comment]
K, I'll try it and i'll be right back.

**Comment by Msrpc** [comment]
@tux-mind IT WORKS :D!!!!! Thank you so much for all the work you have put in.

**Comment by TheRollingEdit** [comment]
Seems to work perfectly. But could you tell me how to cut off someone's device from my network(so they can't use the internet) I can't seem to figure it out.

**Comment by Msrpc** [comment]
@tux-mind how does the Sessions work??? I'm just curious do you need to add a personal metasploit server??

**Comment by tux-mind** [comment]
@TheRollingEdit select the target > mitm > connection killer

@Msrpc sessions are MSF active sessions, learn more about them and you'll understand whet them are.

Also a MSF remote server is needed only if you wish it's a feature that make the MSF work without installing ruby and the MSF on your device. If you have no problem with the installed one you don't need this.

**Comment by Avamander** [comment]
Latest cSploit fixed the issue for me.

**Comment by jonniex7** [comment]
thanks man you are the best keep up the good work!!!

**Comment by Settix** [comment]
Yup can confirm,  no more errors while installing core, ruby or msf ( xperia z3 4.4.4 &  nexus 9 5.0.2)

**Comment by DNBkid** [comment]
unfortunately i`m still getting , error downloading update, and Cannot start process  for MSF : ((

**Comment by Interneedus** [comment]
Reinstall cSploit and try again.

**Comment by DNBkid** [comment]
I have already done it 10 time`s, ill create a logcat  with debug info soon.

**Comment by Ottavio97** [comment]
I downgraded at 1.4.2 and all is fine now. But if you don't update the application, it doesn't let me install core, ruby and MSF. Like if it's all part of an unique upgrade. To avoid future bugs, it will be really good if you can decide to manually install them from Settings Menu. But however, I resolved it by manual in stallation and manual permission settings.
About MSF, Great work for solving core/ruby bug, but I have still the same problem. MSF doesn't want to start. The only difference between then and now is that previously my error was #2 and now is #9, but all the rest is the same. MSF doesn't want to start. I hope you remember first logcat I sent to you, because there's no essential changes now, @tux-mind :)

**Comment by jonniex7** [comment]
huston huston we got a problem but i have faith in you!!!!!
http://pastebin.com/8UM1MTYQ

**Comment by tux-mind** [comment]
@Ottavio97 I added more updates options in settings.

@jonniex7 thanks for the report. fixed in 1.4.7. 

about the recent releases:
> the nightly build server is down, I'm making a lot of releases to make cSploit available to you with the latest changes. will purchase a build server in a far future, when we'll reach more donations.

thanks for your patience.

**Comment by jonniex7** [comment]
&again we(you,all) managed to land down the rocket.... 
                                                     BIG THANKS TO
>>>>>>>>>>>>>>>!!!!!!!!!!!!!!!!!!!!!!!!!--tux-mind--!!!!!!!!!!!!!!!!!!!!!!!!!!!!!<<<<<<<<<<<<<<<<<<<<<<<<<<
            %%%%%%%%%%%%%patient is virtue%%%%%%%%%%%%%%%%%%%%%

**Comment by DNBkid** [comment]
Great news for my huawei p6 i have downloaded 1.4.9 and now Metasploit have downloaded and installed finaly correct.  Big Big Thanks to your great work tux-mind   !!!

**Comment by tux-mind** [comment]
glad to see that you enjoy all my efforts :blush:

**Comment by cesararduino** [comment]
@tux-mind Even after the 1.5.3 version of cSploit, the gem installation of the metasploit framework update fails. Then I try to start metasploit and it says "cannot start process". I tried manually extracting the mfs.zip and putting it in "/data/data/org.csploit.android/files" but when I open cSploit, it asks to update metasploit framework again. And if I say yes, it fails and the extracted folder (mfs) I put in "/data/data/org.csploit.android/files" dissapears. How do I make metasploit work?

**Comment by Systemad** [comment]
@cesararduino csploit should take care of extracting and setting right file permission and other stuff, so manually extracting msf isn't something you should do. the only thing is to wait until it's fixed.

**Comment by cesararduino** [comment]
Oh okay, but I thought that he already made an update to the application and after the update many said that it began working for them. However, the metasploit framework update still doesn't work for me. So, just curious. Is tux-mind going to make another update? And how do I create a logcat so he can see my problem?

**Comment by cesararduino** [comment]
@Systemad

**Comment by ozc** [comment]
Still same problem on Nexus 7 2013, 5.1. Tried both nightly and release versions.

**Comment by 4cylindersoul** [comment]
Im on ver 1.5.3 and its tells me that it cant install msf gems afyer extracting it

**Comment by jonniex7** [comment]
msf error 4.4.2 s3

**Comment by Darrennchan8** [comment]
I have the Same problem... :( 
Here's my logcat>>> http://pastebin.ca/3078990
Here's my log from the data partition>>> http://pastebin.ca/3079004

**Comment by jonniex7** [comment]
may the force be with us!!!!!!!!!

**Comment by GitHubTpac** [comment]
Hey Tux-Mind
msf installing issue.  
running cm11 4.4.4 on galaxy s4
heres my catlog https://ghostbin.com/paste/kmdah

**Comment by tux-mind** [comment]
should be fixed with latest versions. some little bugs still around but thanks to some great new collaborator I'm fixing them all.

**Comment by jonniex7** [comment]
no errors but need some fixes as you know!i tried to install in a intel android tablet but says me no core found!

**Comment by tux-mind** [comment]
yeah @jonniex7 , for now only arm is supported.

**Comment by jonniex7** [comment]
Also now msf rpc daemon not connected and thanks in advance for continuing that great app!!!!

**Comment by tux-mind** [comment]
@jonniex7 what do you mean with `Also now msf rpc daemon not connected` ?
you may refer to #287 , which has been just fixed. if so try out the nightly build in the next hour.

and thank you for testing it out :blush:

**Comment by jonniex7** [comment]
Getting better with each version!! Ty in advance!! Just rpcd is standby and waiting to connect so much time! S3 4.4.2 QS boeffla kernel

**Comment by jonniex7** [comment]
By the way the notification mode of rpcd is brilliant idea!!

**Comment by ft-segfault** [comment]
I keep having this error :"Connection to the MSF RPC daemon FAILED!" even after reinstalling the app, and upgrading Ruby and MSF a few times.
When i try to launch a "session on pwned target" i get this error "MSF RPC not connected".
I'm on CyanogenMod 12.1 ( Android 5.1.1 ).
What should i do to get Metasploit working ?
Thanks
